docs/src/cli/deploy-a-program.md

```diff
@@ -0,0 +1,217 @@
+
+To learn about developing and executing programs on Solana, start with the
+[overview](developing/programming-model/overview.md) and then dig into the
+details of [deployed programs](developing/deployed-programs/overview.md).
+
+To deploy a program, use the Solana tools to interact with the on-chain loader
+to:
+
+- Initialize a program account
+- Upload the program's shared object to the program account's data buffer
+- Verify the uploaded program
+- Finalize the program by marking the program account executable.
+
+Once deployed, anyone can execute the program by sending transactions that
+reference it to the cluster.
+
+## Usage
+
+### Deploy a program
+
+To deploy a program, you will need the location of the program's shared object
+(the program binary .so)
+
+```bash
+solana program deploy <PROGRAM_FILEPATH>
+```
+
+Successful deployment will return the program id of the deployed program, for
+example:
+
+```bash
+Program Id: 3KS2k14CmtnuVv2fvYcvdrNgC94Y11WETBpMUGgXyWZL
+```
+
+Specify the keypair in the deploy command to deploy to a specific program id:
+
+```bash
+solana program deploy --program-id <KEYPAIR_FILEPATH> <PROGRAM_FILEPATH>
+```
+
+If the program id is not specified on the command line the tools will first look
+for a keypair file matching the `<PROGRAM_FILEPATH>`, or internally generate a
+new keypair.
+
+A matching program/keypair file is generated automatically by the program build
+tools and looks like:
+
+```bash
+./path-to-program/program.so
+./path-to-program/program-keypair.json
+```
+
+### Showing a program account
+
+To get information about a deployed program:
+
+```bash
+solana show <ACCOUNT_ADDRESS>
+```
+
+An example output looks like:
+
+```bash
+Program Id: 3KS2k14CmtnuVv2fvYcvdrNgC94Y11WETBpMUGgXyWZL
+ProgramData Address: EHsACWBhgmw8iq5dmUZzTA1esRqcTognhKNHUkPi4q4g
+Authority: FwoGJNUaJN2zfVEex9BB11Dqb3NJKy3e9oY3KTh9XzCU
+Last Upgraded In Slot: 63890568
+Program Length: 5216 (0x1460) bytes
+```
+
+- `Program Id` is the address that can be referenced in an instruction's
+  `program_id` field when invoking a program.
+- `ProgramData Address` is the account associated with the program account that
+  holds the program's data (shared object).
+- `Authority` is the program's upgrade authority.
+- `Last Deployed In Slot` is the slot in which the program was last deployed.
+- `Data Length` is the size of the space reserved for deployments.  The actual
+  space used by the currently deployed program may be less.
+
+### Redeploy a program
+
+A program can be redeployed to the same address to facilitate rapid development,
+bug fixes, or upgrades.  Matching keypair files are generated once so that
+redeployments will be to the same program address.
+
+The command looks the same as the deployment command:
+
+```bash
+solana program deploy <PROGRAM_FILEPATH>
+```
+
+By default, programs are deployed to accounts that are twice the size of the
+original deployment.  Doing so leaves room for further redeployments.  But, if
```

Comment:
Doing so leaves room for further redeployments >
Doing so leaves room for program growth in future redeployments

(Otherwise, it sort of sounds like future redeployments are written after the original in the same account.)